resources/aws_ec2.go

```diff
@@ -971,25 +995,38 @@ func (obj *AwsEc2Res) snsListener(listenAddr string) (net.Listener, error) {
 }
 
 // snsPostHandler listens for posts on the SNS Endpoint.
-// TODO: download pem and check message against signature
 func (obj *AwsEc2Res) snsPostHandler(w http.ResponseWriter, req *http.Request) {
 	if req.Method != "POST" {
 		http.Error(w, "Invalid request method", http.StatusMethodNotAllowed)
 		return
 	}
-	// decode json
+	// Decode the post. If an error is produced we either ignore the post,
+	// or if ErrorJSONStrict is true, send the error through awsChan so
+	// Watch() can return the error and the engine can shut down.
 	decoder := json.NewDecoder(req.Body)
 	var post postData
 	if err := decoder.Decode(&post); err != nil {
-		http.Error(w, "Bad request", http.StatusBadRequest)
```

Comment:
Why did you move this?
My comment about having log earlier was about log vs. http -- not "internal action" vs. "http". I think the http should happen first, and then we shutdown. Yes you can put the log message before that.

Reply:
I figured we wanted one or the other, but that makes sense. Changed.